### PR TITLE
rebrand as Matomo App

### PR DESCRIPTION
**Pull request by Findus23**
TODO:
- [x] replace all mentions of Piwik in code (apart from name of options for backwards compatibility)
- [x] add schema to install.json for validation
- [x] add order to options to make the install.json valid
- [x] add options for DNT and disabling cookies (are there any other commonly used tracking code modifications?)
- [x] update standard.js for code linting
- [x] change links in package.json
- [x] change tracking code to work like the current one (but don't use minified variable names)
- [x] increase version number
- [ ] relicense code: GPL might create huge issues, so should we use MIT license (more popular) or three-clause BSD (like the tracking code)?

Things outside this repo:
- [ ] Rename app to Matomo Analytics
- [ ] replace icon
- [ ] change Piwik and URLs in description text
- [ ] maybe link to this repo in the description text
- [ ] Change name of developer and support contact
- [ ] change link to website and privacy policy
- [ ] change first screenshot (maybe use the one from the Matomo changelogs instead)
- [ ] test if new version of the app still works

### CODE REVIEW

File: install.json [hunk 1]
```diff
@@ -13,19 +14,35 @@
...
+      "disableCookies": {
+        "title": "Disable all tracking cookies",
+        "description": "Disables all first party cookies",
+        "type": "boolean",
+        "default": false,
+        "order": 3
+      },
+      "clientSideDNT": {
+        "title": "Enable client side DoNotTrack detection",
+        "description": "So tracking requests will not be sent if visitors do not wish to be tracked."
```
Comment (MichaIng) [inline install.json]: Trailing comma `,` missing. And I checked back and forth the blow `:` why it is shown red in GitHub viewer 😄.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,2 @@
-# Piwik Analytics Cloud by InnoCraft - Cloudflare App
+# Matomo Analytics Cloud by InnoCraft - Cloudflare App
```
Comment (MichaIng) [inline README.md]: In the `package.json` "InnoCraft" has been replaced with "Matomo Team". Would it make sense to align the README then?

File: install.json (hunk 1 above)
Comment (Findus23) [reply install.json]: you are totally right. I wondered about it too, but then forgot about it. And my editor complained about it, but it was just outside the view.

File: README.md (hunk 2 above)
Comment (Findus23) [reply README.md]: Makes sense.
@mattab, what do you say?

**Comment by MichaIng** [comment]
What do you think about adding switches for `setDomains` and `setCookieDomain` (`*.domain.org`) to enable smoother tracking across sub domains?
```
  let domain = String(options.piwikdomain)

  if (options.trackAcrossSubdomains) {
    _paq.push(["setDomains", ["*." + domain]]);
  }

  if (options.hideSubdomainOutlinks) {
    _paq.push(["setCookieDomain", "*." + domain]);
  }

  if (domain.indexOf('http') !== 0 && domain.indexOf('//') !== 0) {
    domain = 'https://' + domain
  }
```
**_EDIT: Ah, it needs to be assured that the domain does not contain a scheme, no path and no trailing slash etc._**

Another idea I had was to allow customising the whole tracking script snippet, so one can follow the docs to configure it freely. But as I see the way now how Cloudflare apps are structured, it seems to be no trivial task, and, it is probably not a good idea since the app could be misused then to inject any other JavaScript that has nothing to do with Matomo at all.

**Comment by Findus23** [comment]
> Another idea I had was to allow customising the whole tracking script snippet, so one can follow the docs to configure it freely

If I understand the cloudflare docs correctly they explicitly don't allow settings that customize the code:
>  Submitted apps that use code fields to insert script tags will be rejected.
See our ease of use and security guidelines. 
> https://www.cloudflare.com/apps/developer/docs/install-json/options

It would also make sense to propably hide all these options under an advanced settings section as explained in https://www.cloudflare.com/apps/developer/docs/ease-of-use

**Comment by MichaIng** [comment]
> If I understand the cloudflare docs correctly they explicitly don't allow settings that customize the code:

Yes, totally makes sense!

> It would also make sense to propably hide all these options under an advanced settings section

Agreed, especially when more options are added.
_________
Btw, about sub domain related options: Does Cloudflare somehow make global variables available, i.e. the domain that app is installed on? Since the Matomo instance that tracks this domain could be theoretically hosted on a different domain (right?), so deriving the domain that is to be tracked from the domain of the Matomo instance is probably not always correct.

File: README.md (hunk 2 above)
Comment (tsteur) [reply README.md]: That's fine to change @Findus23

Comment (Findus23) [reply README.md]: done

**Comment by MichaIng** [comment]
Anything missing here? While more options could be added, together with an advanced settings section, I think it's okay to do that in subsequent PRs.

File: source/app.js [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
-        domain = 'https://' + domain;
-    }
+  if (options.clientSideDNT) {
+    window._paq.push(['setDoNotTrack', true])
+  }
 
-    if (domain.charAt(domain.length - 1) !== '/') {
-        domain = domain + '/';
-    }
+  let domain = String(options.piwikdomain)
```
Comment (tsteur) [inline source/app.js]: might need to change this to `var` so it works with more browsers

File: source/app.js [hunk 4]
```diff
@@ -1,31 +1,41 @@
 (function () {
-    'use strict';
+  'use strict'
 
-    var options = INSTALL_OPTIONS;
+  const options = INSTALL_OPTIONS
```
Comment (tsteur) [inline source/app.js]: same here, be good to change this to `var`

File: source/app.js [hunk 5]
```diff
@@ -1,31 +1,41 @@
...
-    document.head.appendChild(piwikVendorScript);
-}());
+  if (domain.charAt(domain.length - 1) !== '/') {
+    domain = domain + '/'
+  }
+
+  window._paq.push(['setTrackerUrl', domain + 'matomo.php'])
+  window._paq.push(['setSiteId', options.piwikidsite])
+
+  const piwikVendorScript = document.createElement('script')
```
Comment (tsteur) [inline source/app.js]: also here the const should be replaced by `var`

**Comment by tsteur** [comment]
Left a few comments. @Findus23 is there anything missing from your side? Happy to merge this and update it on the cloudflare marketplace if all still works.

**Comment by Findus23** [comment]
@tsteur Replaced let/const with var

I think everything else should be ready for publishing

**Comment by tsteur** [comment]
This is now done.

> Your app has been successfully submitted to Cloudflare and is awaiting approval. Updates on your submission will be sent to in about 3-4 business days.

Also updated the app store description and some images etc.

Thanks for everyone's help on this!

**Comment by MichaIng** [comment]
Many thanks. I'll give it a try once the 3-4 days have been passed 👍.

**Pull request merged**